src/Microsoft.AspNet.Hosting.Server.Abstractions/IHttpApplication.cs

```diff
@@ -0,0 +1,33 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Threading.Tasks;
+using Microsoft.AspNet.Http.Features;
+
+namespace Microsoft.AspNet.Hosting.Server
+{
+    /// <summary>
+    /// Represents an HttpApplication.
+    /// </summary>
+    public interface IHttpApplication<THttpContext>
```

Comment:
As per offline discussion with @davidfowl, we are going to keep the current naming with the exception of renaming `ProcessRequest` to `ProcessRequestAsync`